tests/test_flask_util.py

```diff
@@ -282,15 +285,18 @@ def test_callback_view_errors(self):
 
         # Code exchange error
         with self.app.test_client() as c:
-            with Http2Mock(status=500):
+            with self.app.test_request_context():
+                # Flask doesn't create a request context with a session
+                # transaction for some reason, so, set up the flow here,
+                # then apply it to the session in the transaction.
+                self.oauth2._make_flow(return_url='/return_url')
                 with c.session_transaction() as session:
-                    session['google_oauth2_csrf_token'] = 'tokenz'
-
-                state = json.dumps({
-                    'csrf_token': 'tokenz',
-                    'return_url': '/return_url'
-                })
+                    session.update(flask.session)
+                    csrf_token = session['google_oauth2_csrf_token']
+                    flow = self.oauth2._get_flow_for_token(csrf_token)
+                    state = flow.params['state']
 
+            with Http2Mock(status=500):
```

Comment:
`six.moves.http_client.INTERNAL_SERVER_ERROR` rather than the literal numeric value 500, please.

Or `httplib.INTERNAL_SERVER_ERROR` as it's known locally in this file, I guess.